clients/cobol-lsp-vscode-extension/src/type/e4eApi.d.ts

```diff
@@ -0,0 +1,175 @@
+};
+
+export type ExternalConfigurationOptions = {
+  compiler: string;
+  preprocessor: string[];
+  type: string;
+};
+
+export const defaultConfigs: ExternalConfigurationOptions = {
+  ...{
+    compiler: "IGYCRCTL",
+    preprocessor: ["DSNHPC", "DFHECP1$"],
+    type: ProcessorConfigurationType.COBOL,
+  },
+};
+
+export interface E4E {
+  isEndevorElement: (uri: string) => boolean;
+  getProfileInfo: (
+    uriStringOrPartialProfile: Partial<ResolvedProfile> | string,
+  ) => Promise<ResolvedProfile | Error>;
+  listElements: (
+    profile: ResolvedProfile,
+    type: {
+      use_map: boolean;
+      environment: string;
+      stage: string;
+      system: string;
+      subsystem: string;
+      type: string;
+    },
+  ) => Promise<[Filename, Fingerprint][] | Error>;
+  getElement: (
+    profile: ResolvedProfile,
+    type: {
+      use_map: boolean;
+      environment: string;
+      stage: string;
+      system: string;
+      subsystem: string;
+      type: string;
+      element: string;
+      fingerprint: string;
+    },
+  ) => Promise<[Content, Fingerprint] | Error>;
+  listMembers: (
+    profile: ResolvedProfile,
+    type: {
+      dataset: string;
+    },
+  ) => Promise<MemberName[] | Error>;
+  getMember: (
+    profile: ResolvedProfile,
+    type: {
+      dataset: string;
+      member: string;
+    },
+  ) => Promise<Content | Error>;
+  getConfiguration: (
+    sourceUri: string,
+    options: ExternalConfigurationOptions,
+  ) => Promise<E4EExternalConfigurationResponse | Error>;
+  onDidChangeElement: vscode.Event<ElementInfo[]>;
+}
+const nameof = <T>(name: keyof T) => name;
+
+export function validateE4E(e4e: any): e4e is E4E {
+  const valid =
+    e4e instanceof Object &&
+    nameof<E4E>("listElements") in e4e &&
+    nameof<E4E>("getElement") in e4e &&
+    nameof<E4E>("listMembers") in e4e &&
+    nameof<E4E>("getMember") in e4e &&
+    nameof<E4E>("isEndevorElement") in e4e &&
+    nameof<E4E>("getProfileInfo") in e4e &&
+    nameof<E4E>("getConfiguration") in e4e &&
+    nameof<E4E>("onDidChangeElement") in e4e;
+  if (!valid) throw Error("incompatible interface");
+  return valid;
+}
+
+export function translateLibs(x: string, profile: string): string;
+export function translateLibs<T extends object>(
+  x: T,
+  profile: string,
+): T & { profile: string };
+export function translateLibs(x: string | object, profile: string) {
+  if (typeof x === "string") return x;
+  return { ...x, profile };
+}
+export function profileAsString(profile: ResolvedProfile) {
+  return `${profile.instance}.${profile.profile}`;
+}
+
+type E4EExternalConfigurationResponse = {
+  pgms: ReadonlyArray<{
+    program: string;
+    pgroup: string;
+    options?: {
+      [key: string]: string;
+    };
+  }>;
+  pgroups: ReadonlyArray<{
+    name: string;
+    libs: (
+      | {
+          member: string;
+          dataset: string;
+        }
+      | {
+          fingerprint: string;
+          element: string;
```

Comment:
The `member`, `fingerprint` and `element` are not provided by the `getConfiguration` function.